### ISSUE TITLE
German translation for "Ed. by" etc. is wrong if editor is an institute or female

### ISSUE THREAD

**Issue by nikonak** [plk/biblatex#1016]
Writing my "Magisterarbeit" I encountered a problem with the German translation for "byeditors" etc. For a `book` entrytype, setting the `editor` field gives "Hrsg. von (editor)". This is fine if the editor is an actual person.

But consider the following MWE:

```tex
\documentclass[ngerman]{article}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{biblatex}

\begin{filecontents}{\jobname.bib}
  @book{Comenius_2018,
    author = {Nicola Bücker and Kirsti Greier},
    gender = {pf},
    editor = {{Comenius-Institut}},
    editorb = {Peter Schreiner},
    editorbtype = {collaborator},
    title = {Gottesdienstliche Angebote mit Kindern},
    subtitle = {Empirische Befunde und Perspektiven},
    langid = {german},
    series = {Evangelische Bildungsberichterstattung},
    number = {1},
    location = {Münster},
    year = {2018}
  }
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}
  \nocite{Comenius_2018}
  \printbibliography
\end{document}
```

Here, `editor` is not an actual person, but an institute. As such, it needs an article (emphasised below).

In the above example, the institutes gender is neutral (and it would be the same for masculine), so it should print "Hrsg. von *dem* Comenius-Institut" or "Hrsg. *vom* Comenius-Institut". If the institute had a feminine gender, it should print e.g. "Hrsg. von *der* Evangelischen Kirche Deutschland".

If a verbose style is used, the same is true for an actual female person: "Anna Musterfrau, Herausgeber" is printed instead of "Anna Musterfrau, Herausgeberin". This also applies to the (admittedly seldomly used, at least in my case) translations for compiler, redactor etc.

Now, I understand that this would require not only expanding the datamodel by e.g. `editor(a,b,c)isnonperson` and `editor(a,b,c)gender` (or only `editor(a,b,c)gender` with distinct fields for people and institute, e.g. `sm`/`ism`, `sn`/`isn` etc.), but also rewriting lots of the current way biblatex handles editors. This is by no means a feature request, but a call to attention as I could imagine the same problem persists in other languages. For anyone encountering this problem, two possible workarounds are:

1. Changing the translations to not give "Hrsg. von (editor)" etc., but "Hrsg: (editor)", "(editor), Hrsg." (similar as for `article`, `incollection` etc.); "Hrsg. v. (editor)" doesn't solve the feminine case, unfortunately.
2. Using the `titleaddon` field to print the desired string verbatim (i.e. in above case setting `titleaddon` to "Hrsg. vom Comenius-Institut" while unsetting `editor`).

**Comment by moewew** [comment]
The "Evangelische Kirche Deutschland" example is actually really tricky, because you'd want

> Evangelische Kirche Deutschland (Hrsg.): ...

but

> Hrsg. von der Evangelischen Kirche Deutschland

from the same input depending on the context or style preferences. So this not only requires not only support from bibstrings and bibmacros, but also different inputs (since I don't think we want to try to have `biblatex` decline field contents).

**Comment by nikonak** [comment]
God forbid! I think that example really is an edge case. It has an academically accepted shorthand (EKD) anyway. But then again, institutes with adjectives in their name are not uncommon at all.

Maybe multiscript support could be facilitated for different inputs? Food for thought.

**Comment by gusbrs** [comment]
For the record, I have a similar case for letters which might be worth mentioning here.

A typical letter reference/citation will start with "John Doe to James Smith", but "John Doe to the Ministry XYZ" if it is some sort of organization/entity. Depending on treatment, a person might also deserve an article: "John Doe to the Director XYZ".

I've dealt with it by creating a `recipientgender` field, constrained to `sf|sm|sn|pf|pm|pn` and by having corresponding bibfields in the languages of interest. I could likely have done something similar with field annotation. But, either way, the style has to support it.

Just another data point.

**Comment by moewew** [comment]
> Maybe multiscript support could be facilitated for different inputs? Food for thought.

Yeah, I think that could work. The dative case could just be a 'version' of the field. Of course it requires user intervention (and some doubling of the input `author = {{Evangelische Kirche Deutschland}}, author_dat = {{der Evangelischen Kirche Deutschland}},`), but then it would probably be fairly straightforward.

With field annotations available it would probably be more natural to give the gender and other necessary information 'institution or not' in a field annotation. (The current `gender` field is a bit unfortunate, since it can apply to `author` or `editor` depending on field availability, but not separately to each of them.) Then this could even be done per item if desirable.
